Tests/UnitTests/Core/Detector/BlueprintHelperTests.cpp

```diff
@@ -0,0 +1,302 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Detector/Blueprint.hpp"
+#include "Acts/Detector/detail/BlueprintHelper.hpp"
+
+#include <exception>
+#include <fstream>
+
+namespace Acts {
+namespace Experimental {
+class IInternalStructureBuilder {};
+}  // namespace Experimental
+}  // namespace Acts
+
+BOOST_AUTO_TEST_SUITE(Experimental)
+
+BOOST_AUTO_TEST_CASE(BlueprintHelperSorting) {
+  // Create  root node
+  std::vector<Acts::BinningValue> detectorBinning = {Acts::binR};
+  std::vector<Acts::ActsScalar> detectorBoundaries = {0., 50., 100.};
+  auto detector = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "detector", Acts::Transform3::Identity(), Acts::VolumeBounds::eCylinder,
+      detectorBoundaries, detectorBinning);
+
+  BOOST_CHECK(detector->parent == nullptr);
+  BOOST_CHECK(detector->children.empty());
+  BOOST_CHECK(detector->name == "detector");
+
+  std::vector<Acts::BinningValue> pixelsBinning = {Acts::binZ};
+  std::vector<Acts::ActsScalar> pixelsBoundaries = {20., 50., 100.};
+
+  auto pixels = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "pixels", Acts::Transform3::Identity(), Acts::VolumeBounds::eCylinder,
+      pixelsBoundaries, pixelsBinning);
+
+  std::vector<Acts::ActsScalar> beamPipeBoundaries = {0., 20., 100.};
+  auto beamPipe = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "beam_pipe", Acts::Transform3::Identity(), Acts::VolumeBounds::eOther,
+      beamPipeBoundaries);
+
+  std::vector<Acts::ActsScalar> gapBoundaries = {20., 50., 10.};
+  auto gap0 = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "gap0", Acts::Transform3::Identity() * Acts::Translation3(0., 0., -90.),
+      Acts::VolumeBounds::eCylinder, gapBoundaries);
+
+  std::vector<Acts::ActsScalar> layerBoundaries = {20., 50., 80.};
+  auto layer = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "layer", Acts::Transform3::Identity(), Acts::VolumeBounds::eCylinder,
+      layerBoundaries,
+      std::make_shared<Acts::Experimental::IInternalStructureBuilder>());
+
+  auto gap1 = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "gap1", Acts::Transform3::Identity() * Acts::Translation3(0., 0., 90.),
+      Acts::VolumeBounds::eCylinder, gapBoundaries);
+
+  // Add the nodes in a random fashion
+  pixels->add(std::move(gap1));
+  pixels->add(std::move(gap0));
+  pixels->add(std::move(layer));
+  // Add pixels and beam pipe in reverse order
+  detector->add(std::move(pixels));
+  detector->add(std::move(beamPipe));
```

Comment:
I like this a lot. I think this can be made a little less boiler-platey from C++, but that's not critical.

Reply:
\The following PRs will showcase this a little bit, in fact - for many use cases we should be able to avoid the boiler plate code, but for the UnitTests I will just create the blueprint by hand.